fundingmanager.go

```diff
@@ -1539,7 +1562,9 @@ func (f *fundingManager) handleFundingAccept(peer lnpeer.Peer,
 	// As they've accepted our channel constraints, we'll regenerate them
 	// here so we can properly commit their accepted constraints to the
 	// reservation.
-	chanReserve := f.cfg.RequiredRemoteChanReserve(resCtx.chanAmt, msg.DustLimit)
+	chanReserve := f.cfg.RequiredRemoteChanReserve(
+		msg.ChannelReserve, resCtx.chanAmt, msg.DustLimit,
```

Comment:
I think this allows the counterparty to control their own `ChannelReserve` - the terminology between "our contribution" / "remote contribution" here always confuses me

Reply:
Updated this, decided to leave the functions alone completely and just overwrite with chan acceptor values in `handleFundingOpen` if they are present. Simpler IMO, and fewer code changes.